synapseclient/models/protocols/agent_protocol.py

```diff
@@ -0,0 +1,163 @@
+"""Protocol for the methods of the Agent and AgentSession classes that have
+synchronous counterparts generated at runtime."""
+
+from typing import TYPE_CHECKING, Optional, Protocol
+
+from synapseclient import Synapse
+
+if TYPE_CHECKING:
+    from synapseclient.models import Agent, AgentSession, AgentSessionAccessLevel
+
+
+class AgentSessionSynchronousProtocol(Protocol):
+    """Protocol for the methods of the AgentSession class that have synchronous counterparts
+    generated at runtime."""
+
+    def start(self, *, synapse_client: Optional[Synapse] = None) -> "AgentSession":
+        """Starts an agent session.
+
+        Arguments:
+            synapse_client: The Synapse client to use for the request.
```

Comment:
To update for consistent verbiage:
```
            synapse_client: If not passed in and caching was not disabled by
                    `Synapse.allow_client_caching(False)` this will use the last created
                    instance from the Synapse class constructor.
```